marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/deleteSavedQuery.sjs

```diff
@@ -0,0 +1,12 @@
+'use strict';
+declareUpdate();
+
+var id;
+
+let savedQuery = cts.search(cts.andQuery([cts.collectionQuery("http://marklogic.com/data-hub/saved-query"),
```

Comment:
For deleting a document, since we know what the URI should be, just do:

    if (cts.exists(cts.documentQuery(uri)) then delete the uri

We only want to use cts.search when we want the document back.

Reply:
Since only the user who saved the query should be able to delete the query, I am adding the following query:
cts.exists(cts.andQuery([cts.documentQuery(docUri), cts.jsonPropertyValueQuery("owner", xdmp.getCurrentUser())]))